twake/backend/node/src/services/previews/services/processing/image.ts

```diff
@@ -0,0 +1,54 @@
+import sharp, { Sharp } from "sharp";
+import { unlink } from "fs/promises";
+import { getTmpFile } from "../../utils";
+import { ThumbnailResult } from "../../types";
+
+export async function generatePreview(
+  inputPaths: string[],
+): Promise<{
+  output: ThumbnailResult[];
+  done: boolean;
+}> {
+  let output: ThumbnailResult[] = [];
+
+  for (const inputPath of inputPaths) {
+    var result: sharp.OutputInfo;
+
+    try {
+      const outputPath = getTmpFile();
+      const inputMetadata = await sharp(inputPath).metadata();
+      const outputFormat = computeNewFormat(inputMetadata);
+
+      result = await sharp(inputPath).resize(outputFormat).toFile(outputPath);
+      output.push({
+        path: outputPath,
+        width: result.width,
+        height: result.height,
+        type: "image/png",
+        size: result.size,
+      });
+    } catch (error) {
+      console.error("there was an error:", error.message);
+      return { output: [], done: false };
+    }
+    await unlink(inputPath);
+  }
+
+  return {
+    output,
+    done: true,
+  };
+}
+
+function computeNewFormat(inputMetadata: sharp.Metadata): { width: number; height: number } {
+  const maxOutputWidth = 300;
+  const maxOutputHeight = 200;
```

Comment:
You must do `maxOutputHeight = options.height || 200` (ajouter options dans les paramètres de computeNewFormat et de generatePreview)